File: LoadCards/Models/Card.cs

public class Card
{
    public string CardCode { get; set; }
    public string CardName { get; set; }
    public int ManaCost { get; set; }
    public int AttackPower { get; set; }
    public int HealthValue { get; set; } 
    public string CardImageLink { get; set; }
    public CardRegions Regions { get; set; }
    public CardTypes Type { get; set; }
    public CardRarity Rarity { get; set; }

    public Card(string cardCode, string cardName, int manaCost, int attackPower, int healthValue, string cardImageLink, CardRegions regions, CardTypes type, CardRarity rarity)
    {
        CardCode = cardCode;
        CardName = cardName;
        ManaCost = manaCost;
        AttackPower = attackPower;
        HealthValue = healthValue;
        CardImageLink = cardImageLink;
        Regions = regions;
        Type = type;
        Rarity = rarity;
    }
}

File: frontend/src/types/Card.ts
import { CardRarity, CardRegions, CardTypes } from "../enums";

type TCard = {
    cardCode: string,
    cardName: string,
    manaCost: number,
    attackPower: number,
    healthValue: number,
    cardImageLink: string,
    cardBackgroundLink: string,
    standard: boolean,
    regions: CardRegions,
    type: CardTypes,
    rarity: CardRarity,
    deckRegions: string
}

export const defaultCard: TCard = {
    cardCode: "",
    cardName: "",
    manaCost: 0,
    attackPower: 0,
    healthValue: 0,
    cardImageLink: "",
    cardBackgroundLink: "",
    standard: false,
    regions: CardRegions.None,
    type: CardTypes.None,
    rarity: 0,
    deckRegions: ""
}

export type { TCard };